src/com/esotericsoftware/yamlbeans/Beans.java

```diff
@@ -232,8 +236,10 @@ public Object get (Object object) throws Exception {
 			try {
 				// The PropertyDescriptor returns the wrong type if the getter is an implementation of a interface method with a
 				// generic return value.
-				type = property.getReadMethod().getDeclaringClass()
-					.getDeclaredMethod(property.getReadMethod().getName(), new Class[0]).getReturnType();
+				Method readMethod = property.getReadMethod().getDeclaringClass()
+					.getDeclaredMethod(property.getReadMethod().getName(), new Class[0]);
+				type = readMethod.getReturnType();
+				genericType = readMethod.getGenericReturnType();

```

Comment:
This code is different in Git now. For a `MethodProperty`, the getter return type can give the generic type. For a field, it can be be gotten this way:
https://github.com/libgdx/libgdx/blob/master/gdx/src/com/badlogic/gdx/utils/reflect/Field.java#L104